recipes/src/recipes/RecipeForm.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import Creatable from 'react-select/creatable';
+
+import styles from '../styles';
+import findRecipe from "./findRecipe";
+
+
+const {
+  Box,
+  FormTitle,
+  StyledForm,
+  FormContent,
+  StyledLabel,
+  StyledInput,
+  StyledTextArea,
+  StyledSubmitButton,
+  StyledCancelButton,
+} = styles;
+
+function RecipeForm(props) {
+  const {
+    recipes,
```

Comment:
FYI, you could avoid having to pass down the recipes (and other props) by extracting them to a context and consuming the context with the `useContext` hook in the component they are needed. May be overkill since there isn't a lot of prop drilling in this code but it's a good pattern when you want to share state between a parent component and its children without having to send the data down via props.

You can also go a step further and create a context hook to extract some of the methods too. This has the benefit of encapsulating the state and methods needed to maintain it in a separate hook file separated from the main component

Reply:
I did actually try using a useReduce hook, but I was running before I could walk, as it failed to update.
So I did it with state and effects - if I had more time I would have put the dispatch function back in and used context for the shared state.